ExtractorUtils/BaseExtractor.cs

```diff
@@ -143,6 +143,37 @@ public virtual async Task Start(CancellationToken token)
                 }
             }
         }
+        
+        /// <summary>
+        /// Method called to add high availability to an extractor.
+        /// </summary>
+        /// <param name="config">Configuration object</param>
+        /// <param name="interval">Optional update state interval.</param>
+        /// <param name="inactivityThreshold">Optional threshold for extractor being inactive.</param>
+        /// <returns></returns>
+        public async Task AddHighAvailability(
+            RawManagerConfig config,
```

Comment:
I'm somewhat uncertain about this method name. It seems to me like it does more than just add high availability, it also waits for activity.

I would also really like it if you could generalize this a bit. I.e. don't take `RawManagerConfig`, take `HighAvailabilityConfig`, then construct a manager based on that. Not entirely sure what the best way to do it is, perhaps create a static `CreateHighAvailabilityManager` method somewhere to check which HA type is specified, then create a manager based on that.